internal/orchestrator/reorg_handler_test.go

```diff
@@ -0,0 +1,502 @@
+	}
+
+	mockMainStorage.EXPECT().LookbackBlockHeaders(big.NewInt(1), mock.Anything, big.NewInt(4)).Return([]common.BlockHeader{
+		{Number: big.NewInt(4), Hash: "hash4a", ParentHash: "hash3a"},
+		{Number: big.NewInt(3), Hash: "hash3a", ParentHash: "hash2"}, // <- fork starts from here
+		{Number: big.NewInt(2), Hash: "hash2", ParentHash: "hash1"},
+		{Number: big.NewInt(1), Hash: "hash1", ParentHash: "hash0"},
+	}, nil)
+
+	forkPoint, err := handler.findFirstForkedBlockNumber(initialBlockHeaders)
+
+	assert.NoError(t, err)
+	assert.Equal(t, big.NewInt(3), forkPoint)
+}
+
+func TestHandleReorg(t *testing.T) {
+	mockRPC := mocks.NewMockIRPCClient(t)
+	mockMainStorage := mocks.NewMockIMainStorage(t)
+	mockOrchestratorStorage := mocks.NewMockIOrchestratorStorage(t)
+
+	mockStorage := storage.IStorage{
+		MainStorage:         mockMainStorage,
+		OrchestratorStorage: mockOrchestratorStorage,
+	}
+
+	mockRPC.EXPECT().GetChainID().Return(big.NewInt(1))
+	mockRPC.EXPECT().GetBlocksPerRequest().Return(rpc.BlocksPerRequestConfig{Blocks: 100})
+	mockRPC.EXPECT().GetFullBlocks(mock.Anything).Return([]rpc.GetFullBlockResult{
+		{BlockNumber: big.NewInt(1), Data: common.BlockData{}},
+		{BlockNumber: big.NewInt(2), Data: common.BlockData{}},
+		{BlockNumber: big.NewInt(3), Data: common.BlockData{}},
+	})
+	mockOrchestratorStorage.EXPECT().GetLastReorgCheckedBlockNumber(big.NewInt(1)).Return(big.NewInt(3), nil)
+
+	reorgStart := big.NewInt(1)
+	reorgEnd := big.NewInt(3)
+
+	mockMainStorage.EXPECT().DeleteBlockData(big.NewInt(1), mock.Anything).Return(nil)
+	mockMainStorage.EXPECT().InsertBlockData(mock.Anything).Return(nil)
+
+	handler := NewReorgHandler(mockRPC, mockStorage)
+	err := handler.handleReorg(reorgStart, reorgEnd)
+
+	assert.NoError(t, err)
+}
+
+func TestStart(t *testing.T) {
+	mockRPC := mocks.NewMockIRPCClient(t)
+	mockMainStorage := mocks.NewMockIMainStorage(t)
+	mockOrchestratorStorage := mocks.NewMockIOrchestratorStorage(t)
+
+	mockStorage := storage.IStorage{
+		MainStorage:         mockMainStorage,
+		OrchestratorStorage: mockOrchestratorStorage,
+	}
+
+	mockRPC.EXPECT().GetChainID().Return(big.NewInt(1)).Maybe()
+	mockOrchestratorStorage.EXPECT().GetLastReorgCheckedBlockNumber(big.NewInt(1)).Return(big.NewInt(2000), nil).Maybe()
+	handler := NewReorgHandler(mockRPC, mockStorage)
+	handler.triggerInterval = 100 // Set a short interval for testing
+
+	mockMainStorage.EXPECT().LookbackBlockHeaders(mock.Anything, mock.Anything, mock.Anything).Return([]common.BlockHeader{
+		{Number: big.NewInt(3), Hash: "hash3", ParentHash: "hash2"},
+		{Number: big.NewInt(2), Hash: "hash2", ParentHash: "hash1"},
+		{Number: big.NewInt(1), Hash: "hash1", ParentHash: "hash0"},
+	}, nil).Maybe()
+
+	mockOrchestratorStorage.EXPECT().SetLastReorgCheckedBlockNumber(mock.Anything, mock.Anything).Return(nil).Maybe()
+
+	go handler.Start()
+
+	// Allow some time for the goroutine to run
+	time.Sleep(250 * time.Millisecond)
+}
```

Comment:
why are we using `.Maybe()`s in this one?

Reply:
Maybe was a way to not check the times it gets called - maybe() allows 0 or more calls. 
Changed it to check the actual number of times (100ms interval with 250ms sleep equals to 2 runs)